scipy/optimize/tests/test_linprog.py

```diff
@@ -777,6 +784,40 @@ def cb(xk, **kwargs):
         assert_(callback_complete[0])
         assert_allclose(last_xk[0], res.x)
 
+    def test_issue_6139(self):
+        # Linprog(method='simplex') fails to find a basic feasible solution
+        # if phase 1 pseudo-objective function is outside the provided tol.
+        # https://github.com/scipy/scipy/issues/6139
+
+        # Note: This is not strictly a bug as the default tolerance determines
+        # if a result is "close enough" to zero and should not be expected
+        # to work for all cases.
+
+        c = np.array([1, 1, 1])
+        A_eq = np.array([[1., 0., 0.], [-1000., 0., - 1000.]])
+        b_eq = np.array([5.00000000e+00, -1.00000000e+04])
+        A_ub = -np.array([[0., 1000000., 1010000.]])
+        b_ub = -np.array([10000000.])
+        bounds = (None, None)
+
+        low_tol = 1e-20
+        res = linprog(
+            c, A_ub, b_ub, A_eq, b_eq,
+            bounds=bounds, options={'tol': low_tol}
```

Comment:
Here and below I would explicitly set `method='self.method'` in case the default algorithm is changed later. Sorry I didn't catch it before.